.clang-format

```
@@ -0,0 +1,6 @@
+---
+Language: Cpp
+BasedOnStyle:  LLVM
```

Nit-picking: all values should be either separated by a single space, or aligned in a column.

```suggestion
BasedOnStyle: LLVM
```

---

We need to add format checker for the .clang-format file :-)

---

okay I will check for it .

---

we can use yamllint to check on build for format of .clang-format

---

```
- name: Install Tools
        run: |
          brew install clang-format yamllint

      - name: Validate .clang-format YAML
        run: |
          yamllint -d "{extends: relaxed, rules: {line-length: {max: 120}}}" .clang-format
```
to check on build